### PR TITLE
feat(#6): :sparkles: impl theme mode switch

### PR DESCRIPTION
**Pull request by huang12zheng**
set null with default in futureProvider

### CODE REVIEW

**Comment by frank06** [comment]
Nice! Thank you.  I will leave some notes in your code

File: lib/main.dart [hunk 1]
```diff
@@ -11,13 +12,18 @@ import 'package:flutter_data/flutter_data.dart';
 
 import 'main.data.dart';
 
-void main() {
+void main() async {
+  final container = ProviderContainer();
+  await container.read(repositoryInitializerProvider.future);
+  await container.read(uiSettingsInitProvider.future);
+
```
Comment (frank06) [inline lib/main.dart:19]: We don't want this as we're blocking the UI during async work. See https://github.com/flutterdata/flutter_data_todos/issues/6

File: lib/main.dart [hunk 2]
```diff
@@ -11,13 +12,18 @@ import 'package:flutter_data/flutter_data.dart';
...
   runApp(
-    ProviderScope(
+    UncontrolledProviderScope(
```
Comment (frank06) [inline lib/main.dart:21]: Why `UncontrolledProviderScope`?

Comment (huang12zheng) [reply lib/main.dart:21]: it is to Inject the container.

File: lib/main.dart [hunk 3]
```diff
@@ -11,13 +12,18 @@ import 'package:flutter_data/flutter_data.dart';
...
+      container: container,
       child: const MyApp(),
-      overrides: [
-        configureRepositoryLocalStorage(clear: false),
-      ],
+      // overrides: [
+      //   configureRepositoryLocalStorage(clear: false),
+      // ],
```
[the remark this answers is not in the document]
Comment (huang12zheng) [reply lib/main.dart:26]: UncontrolledProviderScope don't have overrides

File: lib/models/theme.dart [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+import 'package:flutter_data/flutter_data.dart';
+import 'package:json_annotation/json_annotation.dart';
+import 'package:flutter_data_todos/main.data.dart';
+
+part 'theme.g.dart';
+
+@JsonSerializable()
+@DataRepository([])
+class UISetting extends DataModel<UISetting> {
+  final ThemeMode themeMode;
```
[the remark this answers is not in the document]
Comment (huang12zheng) [reply lib/models/theme.dart:11]: 1. There may be a lot of UI Settings
> Although not here
2. If there is the first touch app, the UISetting  value is null. It need to set it with default value.
But user is not necessarily set until we login.

**Comment by huang12zheng** [comment]
Notice that using AsyncData, which has a lot of layers nested,

File: lib/main.dart [hunk 5]
```diff
@@ -33,17 +28,36 @@ class MyApp extends HookConsumerWidget {
...
+                            onRefresh: () async {
+                              ref.refresh(repositoryInitializerProvider);
+                            },
+                            child: const Home()),
+                      );
+                    });
+                  },
+                  orElse: () =>
+                      const Center(child: CircularProgressIndicator()),
+                );
```
Comment (frank06) [inline lib/main.dart:58]: This can be much simpler, there should be only one `when`.

See https://github.com/flutterdata/flutter_data_todos/issues/6

**Pull request closed without merging**